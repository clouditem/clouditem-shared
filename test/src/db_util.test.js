const db = {any: (...args) => args, one: (...args) => args, many: (...args) => args},
  sut = require('./../../src/db_util.js')(db);
test.each`
  table_input                                 | request_params
  ${['table', 'row1', 'row2', 'row3']}        |  ${{row1: 'row1_val', row2: 'row2_val'}}
  ${['table', 'row1', 'row2', 'row3']}        |  ${{row2: 'row1_val', row3: 'row2_val'}}
  `('$table_input : $request_params', ({table_input, request_params}) => {

  const object = sut(...table_input),
    result = {};

  for (method in object)
    result[method] = object[method]({promised_params: request_params});
  return expect(result).toMatchSnapshot();

});
