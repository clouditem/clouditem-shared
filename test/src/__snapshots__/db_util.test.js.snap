// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`["table", "row1", "row2", "row3"] : {"row1": "row1_val", "row2": "row2_val"} 1`] = `
Object {
  "delete_from": Array [
    "DELETE FROM table WHERE row1 = \${row1} AND row2 = \${row2} RETURNING row1,row2,row3",
    Object {
      "row1": "row1_val",
      "row2": "row2_val",
    },
  ],
  "insert_into": Array [
    "INSERT INTO table (row1,row2) VALUES (\${row1},\${row2}) RETURNING row1,row2,row3",
    Object {
      "row1": "row1_val",
      "row2": "row2_val",
    },
  ],
  "insert_or_update": Array [
    "INSERT INTO table (row1,row2) VALUES (\${row1},\${row2}) RETURNING row1,row2,row3
       ON CONFLICT (row2) DO SET row1 = \${row1}",
    Object {
      "row1": "row1_val",
      "row2": "row2_val",
    },
  ],
  "select_all": Array [
    "SELECT row1,row2,row3  FROM table",
  ],
  "select_where": Array [
    "SELECT row1,row2,row3 FROM table WHERE row1 = \${row1} AND row2 = \${row2}",
    Object {
      "row1": "row1_val",
      "row2": "row2_val",
    },
  ],
}
`;

exports[`["table", "row1", "row2", "row3"] : {"row2": "row1_val", "row3": "row2_val"} 1`] = `
Object {
  "delete_from": Array [
    "DELETE FROM table WHERE row2 = \${row2} AND row3 = \${row3} RETURNING row1,row2,row3",
    Object {
      "row2": "row1_val",
      "row3": "row2_val",
    },
  ],
  "insert_into": Array [
    "INSERT INTO table (row2,row3) VALUES (\${row2},\${row3}) RETURNING row1,row2,row3",
    Object {
      "row2": "row1_val",
      "row3": "row2_val",
    },
  ],
  "insert_or_update": Array [
    "INSERT INTO table (row2,row3) VALUES (\${row2},\${row3}) RETURNING row1,row2,row3
       ON CONFLICT (row2) DO SET row3 = \${row3}",
    Object {
      "row2": "row1_val",
      "row3": "row2_val",
    },
  ],
  "select_all": Array [
    "SELECT row1,row2,row3  FROM table",
  ],
  "select_where": Array [
    "SELECT row1,row2,row3 FROM table WHERE row2 = \${row2} AND row3 = \${row3}",
    Object {
      "row2": "row1_val",
      "row3": "row2_val",
    },
  ],
}
`;
